### PR TITLE
Feature/hilt

### PR DESCRIPTION
**Pull request by AdrianUlrich**

### CODE REVIEW

**Review by albastardoto — APPROVED** [review]
Good changes but needs better commit messages especially for changes that aren't specifically about the feature the branch is named after.

File: app/src/main/java/com/github/sdpsharelook/authorization/SignUpActivity.kt [hunk 1]
```diff
@@ -170,7 +170,7 @@ class SignUpActivity : AppCompatActivity() {
         val email = findViewById<EditText>(R.id.email).text.toString()
         val password = findViewById<EditText>(R.id.password).text.toString()
 
-        GlobalScope.launch {
+        CoroutineScope(Dispatchers.IO).launch {
```
[the remark this answers is not in the document]
Comment (AdrianUlrich) [reply app/src/main/java/com/github/sdpsharelook/authorization/SignUpActivity.kt:173]: It was already bugfixed once (181c90b0a0b4e75601240a000894d6ce2ae16adb) and merged into main, but someone reverted (0c4151d74212e0005380e089921f0e11f43acc75) it by being bad at merging their own branch

File: app/src/main/java/com/github/sdpsharelook/camera/CameraActivity.kt [hunk 2]
```diff
@@ -92,11 +92,11 @@ class CameraActivity : AppCompatActivity() {
         return permission1
     }
 
-    fun createImage(): File {
-        val timeStamp = SimpleDateFormat("yyyyMMdd_HHmmss").format(Date())
+    private fun createImage(): File {
+        val timeStamp = SimpleDateFormat("yyyyMMdd_HHmmss",Locale.US).format(Date())
```
[the remark this answers is not in the document]
Comment (AdrianUlrich) [reply app/src/main/java/com/github/sdpsharelook/camera/CameraActivity.kt:96]: You're right, it should depend on country. However that's not my job to figure out, I just fixed errors caused by the fact that no locale was being given at all

**Review by rombirli — APPROVED** [review]
Good work !

**Pull request merged**